src/boss/rbo3/rbo3.c

```diff
@@ -531,7 +531,186 @@ void func_us_801922EC(Entity* self) {
+
+void func_us_801923DC(Entity* self) {
+    const int PrimCount = 13;
+    s32 i;
+    s32 primIndex;
+    s16 angle;
+    s32 xOffset;
+    s32 yOffset;
+    s32 x;
+    s32 y;
+    Primitive* prim;
+    s16* offsets;
+    Entity* next;
+    s32 tempY;
+    Entity* player;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(D_us_801804A4);
+        primIndex = g_api.AllocPrimitives(PRIM_GT4, PrimCount);
+        if (primIndex == -1) {
+            DestroyEntity(self);
+            return;
+        }
+
+        self->flags |= FLAG_HAS_PRIMS;
+        self->primIndex = primIndex;
+        prim = &g_PrimBuf[primIndex];
+        self->ext.prim = prim;
+
+        for (i = 0; i < PrimCount; i++) {
+            prim->tpage = 0x1A;
+            prim->clut = 0x194;
+
+            prim->u0 = prim->u1 = i * 4 + 144;
+            prim->u2 = prim->u3 = i * 4 + 148;
+            prim->v0 = prim->v2 = 208;
+            prim->v1 = prim->v3 = 192;
+            prim->r0 = prim->g0 = prim->b0 = 192 - (i * 16);
+
+            LOW(prim->r1) = LOW(prim->r0);
+            LOW(prim->r2) = LOW(prim->r0);
+            LOW(prim->r3) = LOW(prim->r0);
+
+            prim->priority = self->zPriority + 1;
+            prim->drawMode = DRAW_TPAGE2 | DRAW_TPAGE | DRAW_COLORS |
+                             DRAW_UNK02 | DRAW_TRANSP;
+
+            prim = prim->next;
+        }
+
+        offsets = &self->ext.medusaUnk1A.offsets;
+        for (i = 0; i < (PrimCount + 1); i++) {
+            offsets[0] = self->posX.i.hi;
+            offsets[1] = self->posY.i.hi;
+            offsets += 2;
+        }
+
+        angle = self->rotZ;
+        self->velocityX = rcos(angle) * 160;
+        self->velocityY = rsin(angle) * 160;
+        self->ext.factory.unk82 = angle;
+        self->ext.factory.unk80 = 0x40;
+
+    case 1:
+        MoveEntity();
+        player = &PLAYER;
+        angle = GetAngleBetweenEntities(self, player);
+        tempY = 96 - self->ext.factory.unk80;
+        angle = GetNormalizedAngle(tempY, self->ext.factory.unk82, angle);
+        self->velocityX = rcos(angle) * 160;
+        self->velocityY = rsin(angle) * 160;
+        self->ext.factory.unk82 = angle;
+
+        if (!--self->ext.factory.unk80 || self->hitFlags & 0x80) {
+            self->step++;
+        }
+        break;
+
+    case 2:
+        MoveEntity();
+        xOffset = self->posX.i.hi + g_Tilemap.scrollX.i.hi;
+        yOffset = self->posY.i.hi + g_Tilemap.scrollY.i.hi;
+        if ((xOffset < -64) || (yOffset < -64) || (xOffset > 576) ||
+            (yOffset > 320)) {
+            DestroyEntity(self);
+            return;
+        }
+        break;
+    }
+
+    if (!(g_Timer & 7)) {
+        PlaySfxPositional(SFX_ELECTRICITY);
+    }
+
+    // this is the same as self->ext.unkB8. for this
+    // entity 0xB8 is not an entity pointer but the
+    // last pair of offsets.
+    offsets = &self->ext.medusaUnk1A
+                   .offsets[sizeof(self->ext.medusaUnk1A.offsets) / 2];
```

Comment:
The last 4 bytes of `struct Entity` were recently replaced with `struct Entity* unkB8`, but in this case those bytes are used for primitive offsets. That might not be enough of an argument to revert that change, but I'm calling it out in case there end up being other entities end up using that field for something else.